File: leetcode/1592-rearrange-spaces-between-words/reorderSpaces.ts
function reorderSpaces(text: string): string {
  const tempText = text.split(" ");
  const words = tempText.filter((item) => item !== "");
  const wordsCount = words.length;
  const spaceCount = text.length - words.join("").length;
  if (wordsCount <= 1) {
    return `${words.join("")}${" ".repeat(spaceCount)}`;
  }

  const betweenSpaceCount = Math.floor(spaceCount / (wordsCount- 1));
  const restSpace = spaceCount % (wordsCount- 1);

  return `${words.join(" ".repeat(betweenSpaceCount))}${" ".repeat(restSpace)}`;
}

export default reorderSpaces;

File: leetcode/1592-rearrange-spaces-between-words/reorderSpaces.go
package leetcode

import (
	"fmt"
	"strings"
)

func reorderSpaces(text string) string {
	var words = strings.Fields(text)
	var wordsCount = len(words)
	var spaceCount = strings.Count(text, " ")

	if wordsCount <= 1 {
		return fmt.Sprintf("%s%s", strings.Join(words, ""), strings.Repeat(" ", spaceCount))
	}

	var betweenSpaceCount = spaceCount / (wordsCount - 1)
	var restSpace = spaceCount % (wordsCount - 1)

	return fmt.Sprintf("%s%s", strings.Join(words, strings.Repeat(" ", betweenSpaceCount)), strings.Repeat(" ", restSpace))
}
